python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,285 @@
+                for actuator in self.actuators_to_test
+                if self.has_secondary_actuator(actuator)
+            ]
+        )
+
+        # Setting metadata
+        metadata.duration = self.time_one_bump * (
+            len(self.actuators_to_test) + total_tested_secondary
+        )
+
+    def has_secondary_actuator(self, actuator_id: int) -> bool:
+        """Determines whether a given actuator has a
+        secondary axis or not.
+        """
+
+        return actuator_id in self.m1m3_secondary_actuator_ids
+
+    async def run(self):
+        await self.assert_feasibility()
+        start_time = time.monotonic()
+
+        # Get M1M3 detailed state
+        detailed_state = DetailedState(
+            (
+                await self.mtcs.rem.mtm1m3.evt_detailedState.aget(
+                    timeout=self.mtcs.fast_timeout,
+                )
+            ).detailedState
+        )
+        self.log.info(f"Current M1M3 detailed state: {detailed_state!r}.")
+
+        # Put M1M3 in engineering mode
+        await self.mtcs.enter_m1m3_engineering_mode()
+
+        failed_actuators = []
+        failed_primary_index = []
+        failed_secondary_index = []
+        for i, actuator_id in enumerate(self.actuators_to_test):
+            await self.mtcs.assert_all_enabled()
+            # await self.checkpoint(f"Running bump test on FA ID:
+            # {actuator_id}.")
+
+            # Determine if actuator has a secondary actuator
+            secondary_exist = self.has_secondary_actuator(actuator_id)
+
+            # Checkpoint
+            primary_index = self.m1m3_actuator_ids.index(actuator_id)
+            if secondary_exist:
+                secondary_index = self.m1m3_secondary_actuator_ids.index()
+                await self.checkpoint(
+                    f"Running bump test on DAA FA ID: {actuator_id} "
+                    f"(Primary/Secondary: {primary_index}/{secondary_index})"
+                )
+            else:
+                await self.checkpoint(
+                    f"Running bump test on SAA FA ID: {actuator_id} "
+                    f"(Primary {primary_index})"
+                )
+
+            try:
+                await self.mtcs.run_m1m3_actuator_bump_test(
+                    actuator_id=actuator_id,
+                    primary=True,
+                    secondary=secondary_exist,
+                )
+            # raise a RuntimeError if the bump test fails
+            # and continue loop to next actuator_id
+            except RuntimeError:
+                failed_actuators.append(actuator_id)
+                self.log.exception(f"Failed to run bump test on FA ID {actuator_id}.")
+
+            # Getting test status
+            (
+                fa_id,
+                primary_status,
+                secondary_status,
+            ) = await self.mtcs.get_m1m3_bump_test_status()
+
+            # BumpTest.PASSED = 6
+            # BumpTest.FAILED = 7
+            if not secondary_exist:
```

Comment:
This will not add failed primary if there is secondary. There is always primary. So the test shall be:

```
               # Getting primary indexes for failures
                if primary_status == BumpTest.FAILED:
                    failed_primary_index.append(primary_index)
                if secondary_exists:
                    if secondary_status == BumpTest.FAILED:
                        failed_secondary_index.append(secondary_index)
```

but even better (to identify), if you append both index and id, like that:
```
failed_secondary.append(actuator_id, secondary_index)
```